src/Altinn.AccessManagement.Integration/Clients/Altinn2RightsClient.cs

```diff
@@ -0,0 +1,83 @@
+        private readonly ILogger _logger;
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AltinnRolesClient"/> class
+        /// </summary>
+        /// <param name="httpClient">HttpClient from default httpclientfactory</param>
+        /// <param name="sblBridgeSettings">the sbl bridge settings</param>
+        /// <param name="logger">the logger</param>
+        public Altinn2RightsClient(HttpClient httpClient, IOptions<SblBridgeSettings> sblBridgeSettings, ILogger<AltinnRolesClient> logger)
+        {
+            _sblBridgeSettings = sblBridgeSettings.Value;
+            _logger = logger;
+            _client = httpClient;
+            _serializerOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        /// <inheritdoc />
+        public async Task<DelegationCheckResponse> PostDelegationCheck(int userId,RightsDelegationCheckRequest request)
+        {
+            try
+            {
+                string partyId = request.From[0].Value;
+                string serviceCode = request.Resource.FirstOrDefault(r => r.Id == "urn:altinn:servicecode").Value;
+                string serviceEditionCode = request.Resource.FirstOrDefault(r => r.Id == "urn:altinn:serviceeditioncode").Value;
+
+                UriBuilder uriBuilder = new UriBuilder($"{_sblBridgeSettings.BaseApiUrl}authorization/api/rights/delegation/userdelegationcheck?userId={userId}&partyId={partyId}&serviceCode={serviceCode}&serviceEditionCode={serviceEditionCode}");
+                _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                HttpResponseMessage response = await _client.GetAsync(uriBuilder.Uri);
+                string rights = await response.Content.ReadAsStringAsync();
+
+                DelegationCheckResponse delegationCheckResponse = new DelegationCheckResponse();
+
+                if (response.StatusCode == System.Net.HttpStatusCode.OK)
+                {
+                    delegationCheckResponse.RightDelegationCheckResults = JsonSerializer.Deserialize<List<RightDelegationCheckResult>>(rights, _serializerOptions);
+                    return delegationCheckResponse;
+                }
+
+                SBLUserDelegationCheckError validationError = JsonSerializer.Deserialize<SBLUserDelegationCheckError>(rights, _serializerOptions);
+                if (validationError.ModelState == null)
+                {
+                    delegationCheckResponse.Errors.Add("SBLBridge", "Unexpected error from AccessManagement // Altinn2RightsClient // PostDelegationCheck");
+                    return delegationCheckResponse;
+                }
+
+                delegationCheckResponse.Errors.Add(validationError.ModelState.First().Key, validationError.ModelState.First().Value.First());
```

Comment:
In stead of First this could be written as a foreach, and would avoid potential problems if empty

Reply:
Replaced this with a foreach